packages/paste-core/components/disclosure/src/Disclosure.tsx

```diff
@@ -26,23 +26,27 @@ export interface DisclosureProps extends DisclosurePrimitiveInitialState {
   variant?: Variants;
 }
 
-const Disclosure: React.FC<DisclosureProps> = ({children, variant = 'default', state, ...props}) => {
-  const disclosure = state || useDisclosurePrimitiveState({...props});
-  const disclosureContext = {
-    disclosure,
-    variant,
-  };
-
-  if (variant === 'contained') {
-    return (
-      <DisclosureContext.Provider value={disclosureContext}>
-        <Card padding="space0">{children}</Card>
-      </DisclosureContext.Provider>
-    );
+const Disclosure = React.forwardRef<HTMLDivElement, DisclosureProps>(
+  ({children, variant = 'default', state, ...props}, ref) => {
+    const disclosure = state || useDisclosurePrimitiveState({...props});
+    const disclosureContext = {
+      disclosure,
+      variant,
+    };
+
+    if (variant === 'contained') {
+      return (
+        <DisclosureContext.Provider value={disclosureContext}>
+          <Card padding="space0" ref={ref}>
+            {children}
+          </Card>
+        </DisclosureContext.Provider>
+      );
+    }
+
+    return <DisclosureContext.Provider value={disclosureContext}>{children}</DisclosureContext.Provider>;
```

Comment:
Hmm, is it fine that ref isn't passed for variant !== contained?

Reply:
Good catch. I just missed that.